src/app/api/partnerDetails/[userId]/route.ts

```diff
@@ -0,0 +1,93 @@
+import { authenticationError, notFoundError, authorizationError, argumentError } from "@/util/responses";
+import { auth } from "@/auth";
+import { NextResponse, NextRequest } from "next/server";
+import { z } from "zod";
+import { zfd } from "zod-form-data"; // <-- Added import from zod-form-data
+import { PrismaClient } from "@prisma/client"; 
+
+// new Prisma Client
+const prisma = new PrismaClient();
+
+// Zod schema
+const PartnerDetailsFormSchema = zfd.formData({
+  numberOfPatients: zfd.numeric(z.number().min(1, "Number of patients must be positive")),
+  organizationType: zfd.text(z.enum(["NON_PROFIT", "FOR_PROFIT", "RELIGIOUS"])),
+});
+
+/**
+ * Updates a user's partner details.
+ * Accepts parameters as FormData.
+ *
+ * @param {NextRequest} req The incoming request object.
+ * @param {Object} context The route parameters containing the user ID.
+ *
+ * @returns {NextResponse} 401 if request is not authenticated.
+ * @returns {NextResponse} 403 if the session user is a PARTNER modifying another user's details.
+ * @returns {NextResponse} 400 if the request body is invalid.
+ * @returns {NextResponse} 404 if no PartnerDetails record is found.
+ * @returns {NextResponse} 200 with the updated partner details.
+ */
```

Comment:
nit: can you reformat to match [the example](https://www.notion.so/gtbitsofgood/Writing-a-Query-Mutation-182bd5d1ba15805388b2c705056ed4a7?pvs=4#182bd5d1ba15800c83d1d38d02caa22a)